packages/cli/src/credentials/credentials.service.ts

```diff
@@ -37,33 +37,61 @@ export class CredentialsService {
 		});
 	}
 
-	static async getMany(user: User, options?: { disableGlobalRole: boolean }) {
-		type Select = Array<keyof ICredentialsDb>;
+	private static toFindManyOptions(listQueryOptions?: ListQuery.Options) {
+		const findManyOptions: FindManyOptions<CredentialsEntity> = {};
 
-		const select: Select = ['id', 'name', 'type', 'nodesAccess', 'createdAt', 'updatedAt'];
+		type Select = Array<keyof CredentialsEntity>;
 
-		const relations = ['shared', 'shared.role', 'shared.user'];
+		const defaultRelations = ['shared', 'shared.role', 'shared.user'];
+		const defaultSelect: Select = ['id', 'name', 'type', 'nodesAccess', 'createdAt', 'updatedAt'];
 
-		const returnAll = user.globalRole.name === 'owner' && options?.disableGlobalRole !== true;
+		if (!listQueryOptions) return { select: defaultSelect, relations: defaultRelations };
 
-		const addOwnedByAndSharedWith = (c: CredentialsEntity) =>
-			Container.get(OwnershipService).addOwnedByAndSharedWith(c);
+		const { filter, select, take, skip } = listQueryOptions;
+
+		if (filter) findManyOptions.where = filter;
```

Comment:
Only problem I see here is that the name is querying for the exact value and cannot accept partial parts of the name, this is a bit annoying WDYT?

Reply:
I can't remember whether we had the same issue for workflows, but partial name search should be a requirement for both

Reply:
Thank you, added. Workflow already [supports this](https://github.com/n8n-io/n8n/blob/2cfa6d344e2ca95a4f167dd7bc71eebe522e71f3/packages/cli/src/workflows/workflows.services.ts#L141).